icu4c/source/test/cintltst/cloctst.c

```diff
@@ -6747,70 +6747,83 @@ static void TestBug20149() {
 
 static void TestUsingDefaultWarning() {
     UChar buff[256];
-    char cbuff[256];
+    UChar expectedBuff[256];
+    char errorOutputBuff[256];
     UErrorCode status = U_ZERO_ERROR;
     const char* language = "jJj";
+    const char expectedLanguage[] = "jjj";
+    u_charsToUChars(expectedLanguage, expectedBuff, UPRV_LENGTHOF(expectedLanguage));
     int32_t length = uloc_getDisplayLanguage(language, "de", buff, 256, &status);
     if (status != U_USING_DEFAULT_WARNING ||
-        u_strcmp(buff, CharsToUChars("jjj")) != 0 ||
+        u_strcmp(buff, expectedBuff) != 0 ||
```

Comment:
Can we not use `u"jjj"` in cintltst? As long as we avoid using char16_t?

Reply:
compiler of C won't take u""

Reply:
Oh, that would be nice... but as Frank hints at, support for the `char16_t` type (and the `u""` prefix for string literals) was added in C11. We can't use it, since the ICU code must be C99.

Reply:
My memory is that `u"abc"` works but we either cannot use the type `char16_t` or cannot include the relevant header file, or both, because I think the Apple clang compiler doesn't support it.

So then I grepped for `u"` in cintltst/*.c files and found dozens of examples where we already use UTF-16 string literals in the C unit test code. For example: https://github.com/unicode-org/icu/blob/master/icu4c/source/test/cintltst/cdateintervalformattest.c#L328

So please use that for string literals instead of calling conversion functions.

Reply:
Interesting... AFAICT, the `char16_t` type and the `u""` literals are C11 only. (Ex: https://en.cppreference.com/w/c/language/string_literal , https://en.wikipedia.org/wiki/C11_(C_standard_revision)#Changes_from_C99 ).
However, if we're *already* using them in the test code, then I definitely agree that would be *much* cleaner than having to call the conversion functions!

(I guess it sort of seems like we sort of have a mix of C99 and C11 code... Perhaps the `u""` literals are allowed in test code maybe?)

I'll force-push an update to remove the conversion functions in a moment...